src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+                "info and entry_name type") in str(error_exec)
+        await self.teardown_method()
+
+        await self.setup_method()
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean', category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': 'boolean type with default False ',
+                                      'type': 'boolean',
+                                      'default': None
+                                  }})
+        assert ("TypeError: entry_val must be a string for item_name info " +
+                "and entry_name default") in str(error_exec)
+        await self.teardown_method()
+
+    async def test_set_category_item_value_entry(self):
+        """
+        Test updating of configuration.value for a specific key
+        :assert:
+            1. `default` and `value` in configuration.value are the same
+            2. `value` in configuration.value gets updated, while `default` does not
+        """
+        stmt = sa.select([_configuration_tbl.c.value]).select_from(_configuration_tbl).where(
+            _configuration_tbl.c.key == 'boolean')
+        await self.setup_method()
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}})
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(stmt):
+                        assert result[0]['info']['value'] == 'False'
+                        assert result[0]['info']['default'] == 'False'
+        except Exception:
+            raise
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(stmt):
+                        assert result[0]['info']['value'] == 'True'
+                        assert result[0]['info']['default'] == 'False'
+        except Exception:
+            raise
+        await self.teardown_method()
+
+    async def test_set_category_item_value_error(self):
+        """
+        Test updating of configuration.value when configuration is not set
+        :assert:
+            As of now the expected behavior is nothing gets returned; however,
+            there is an open issue expecting an error be returned when an update
```

Comment:
open issue? better to skip or xFail (expected failure) this test with JIRA id.
`@pytest.mark.skip(reason="FOGL-xxx")
` async def test_set_category_item_value_error(self):):`